purchase_date_planned_manual/models/purchase_order.py

```diff
@@ -9,6 +9,21 @@
 from odoo.fields import Datetime as Dt
 
 
+class PurchaseOrder(models.Model):
+    _inherit = "purchase.order"
+
+    def _compute_date_planned(self):
+        """ date_planned = the earliest date_planned across all order lines. """
+        for order in self:
+            if order.date_planned:
+                continue
+
+            dates_list = order.order_line.filtered(lambda x: not x.display_type and x.date_planned).mapped('date_planned')
+            if dates_list:
+                order.date_planned = min(dates_list)
+            else:
+                order.date_planned = False
```

Comment:
what if the new PO line is not an associated container-deposit product?

From what I understand, the issue of the card is
1. When creating PO, add a PO Line, whose product has container-deposit product (simple an extra product automatically attached).
2. Now, the PO has 2 PO lines, theoretically have same expected arrival date, but they do now. This is the issue

With that being said, a better idea is to create a glue module between this module and `product_packaging_container_deposit`, you can add the condition there.

btw, completely overridden method should be avoided in most cases imo.